lib/features/interactive.js

```diff
@@ -0,0 +1,151 @@
+import assert from 'assert';
+import Debug from 'debug';
+import {DockerExecServer} from 'docker-exec-websocket-server';
+import fs from 'fs-ext';
+import http from 'http';
+import https from 'https';
+import path from 'path';
+import Promise from 'promise';
+import slugid from 'slugid';
+import SharedFileLock from '../shared_file_lock';
+import url from 'url';
+let debug = Debug('docker-worker:features:interactive');
+
+//Number of attempts to find a port before giving up
+const MAX_RANDOM_PORT_ATTEMPTS = 20;
+
+export default class WebsocketServer {
+  constructor () {
+    let id = slugid.v4();
+    this.path = '/' + id;
+    this.lock = path.join('/tmp/', id + '.lock');
+  }
+
+  async link(task) {
+    this.lockFile = await Promise.denodeify(fs.open)(this.lock, 'w');
+
+    return {
+      binds: [{
+        source: path.join(__dirname, '../../bin-utils'),
+        target: '/.taskcluster_utils',
```

Comment:
Hmm... Actually looking at docker, I see that it already has a `/.dockerenv` and `/.dockerinit` files.
How about we follow that convention and make this `/.taskclusterutils/`. Then we also avoid any complications with dash and underscore...